plugin/src/main/java/net/dzikoysk/funnyguilds/feature/placeholders/Placeholders.java

```diff
@@ -0,0 +1,138 @@
+                .bracket("MEMBERS",             guild -> {
+                    String text = Joiner.on(", ").join(UserUtils.getOnlineNames(guild.getMembers())).toString();
+
+                    return ONLINE
+                            .toFormatter(ChatColor.getLastColors(text.split("<online>")[0]))
+                            .format(text);
+                })
+                .bracket("DEPUTIES",            guild -> guild.getDeputies().isEmpty()
+                        ? "Brak"
+                        : Joiner.on(", ").join(UserUtils.getNamesOfUsers(guild.getDeputies())))
+                .bracket("REGION-SIZE",         guild -> config.regionsEnabled ? guild.getRegion().getSize() : messages.gRegionSizeNoValue)
+                .bracket("GUILD-PROTECTION",    bindGuildProtection)
+                .bracket("POINTS-FORMAT",       guild -> IntegerRange.inRangeToString(guild.getRank().getAveragePoints(), config.pointsFormat))
+                .bracket("POINTS",              guild -> guild.getRank().getAveragePoints())
+                .bracket("KILLS",               guild -> guild.getRank().getKills())
+                .bracket("DEATHS",              guild -> guild.getRank().getDeaths())
+                .bracket("ASSISTS",             guild -> guild.getRank().getAssists())
+                .bracket("LOGOUTS",             guild -> guild.getRank().getLogouts())
+                .bracket("KDR",                 guild -> String.format(Locale.US, "%.2f", guild.getRank().getKDR()))
+                .bracket("VALIDITY",            guild -> config.dateFormat.format(new Date(guild.getValidity())))
+                .bracket("LIVES",               Guild::getLives)
+                .bracket("RANK",                guild -> guild.getMembers().size() >= config.minMembersToInclude
+                        ? guild.getRank().getPosition()
+                        : messages.minMembersToIncludeNoValue)
+                .bracket("ALLIES",              guild -> guild.getAllies().isEmpty()
+                        ? messages.alliesNoValue
+                        : Joiner.on(", ").join(GuildUtils.getNames(guild.getAllies())))
+                .bracket("ALLIES-TAGS",         guild -> guild.getAllies().isEmpty()
+                        ? messages.alliesNoValue
+                        : Joiner.on(", ").join(GuildUtils.getTags(guild.getAllies())))
+                .bracket("ENEMIES",             guild -> guild.getEnemies().isEmpty()
+                        ? messages.enemiesNoValue
+                        : Joiner.on(", ").join(GuildUtils.getNames(guild.getEnemies())))
+                .bracket("ENEMIES-TAGS",        guild -> guild.getEnemies().isEmpty()
+                        ? messages.enemiesNoValue
+                        : Joiner.on(", ").join(GuildUtils.getTags(guild.getEnemies())));
+    }
+
+    private final Map<String, Function<T, String>> placeholders = new HashMap<>();
+
+    public Placeholders() {
```

Comment:
To nie powinno być private?

Reply:
Nie, w takim przypadku nie będzie możliwości stworzenia instancji tej klasy, co wyklucza dodawania customowych placeholderów nieokreślonych w statycznych fieldach.

Reply:
To ma sens